File: 工厂模式/工厂模式/ViewController.m
//
//  ViewController.m
//  工厂模式
//
//  Created by qrh on 2019/7/24.
//  Copyright © 2019 zdq. All rights reserved.
//

#import "ViewController.h"
#import "SXComputerFactory.h"
#import "HWComputerFactory.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    id<ComputerFactoryProtocol>  factory = [SXComputerFactory new];
    id<ComputerProtocol> computer = [factory getComputer];
    [computer CPU];
    
    
    id<ComputerFactoryProtocol>  factory1 = [HWComputerFactory new];
    id<ComputerProtocol> computer1 = [factory1 getComputer];
    [computer1 CPU];

}


@end

File: 迭代器模式/迭代器模式/ViewController.swift
//
//  ViewController.swift
//  迭代器模式
//
//  Created by qrh on 2019/7/27.
//  Copyright © 2019 zdq. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        let aggregrate = ConcreteAggregrate<Int>()
        aggregrate.add(10)
        aggregrate.add(12)
        aggregrate.add(15)
        
        let iteator = aggregrate.iterator()
        while iteator.hasNext() {
            print("\(String(describing: iteator.next()))")
        }
    }
}

